### PR TITLE
[i18n/Audio] Enable for pre-generated election date strings and audio

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

Noticed that the only date we need to translate and/or generate audio for, so far, is the election date -- to avoid having to piece together audio fragments to read out dates, this PR adds support for pre-generating language-specific date strings and audio for the election date.

The CDF doesn't currently provide support for `InternationalizedText` strings for the election date, so the approach here is to add a new `vxElectionStrings.json` file to the ballot package, which will contain non-CDF election-specific translations. 

Also adding an `ElectionStringKey.ELECTION_DATE` key and corresponding string extractor and renderer.

## Testing Plan
- New test cases for new ballot package file and string extraction logic
- Updated affected tests

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- ~[ ] I have added a screenshot and/or video to this PR to demo the change~
- ~[ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

**Comment by kofi-q** [comment]
Moving to draft for now (sorry for the thrash) - Jonah mentioned there's a possibility that we can't add custom fields to the CDF.

Going to move this string to a different location in the ballot package and re-open

**Comment by arsalansufi** [comment]
No worries about thrash! I noticed the Slack thread before starting review - makes sense to rethink

**Comment by kofi-q** [comment]
@arsalansufi - LMK what you think about the new approach. Hoping it doesn't complicate the ballot package generation process, but open to any alternatives or tweaks if necessary.

File: libs/types/src/ballot_package.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export enum BallotPackageFileName {
   METADATA = 'metadata.json',
   SYSTEM_SETTINGS = 'systemSettings.json',
   UI_STRING_AUDIO_IDS = 'uiStringAudioIds.json',
+  VX_ELECTION_STRINGS = 'vxElectionStrings.json',
```
[the remark this answers is not in the document]
Comment (kofi-q) [reply libs/types/src/ballot_package.ts:20]: Ooh, I really like option 1 as a generalised solution for our custom fields -- agreed that that might need a bit of planning/hashing out details, but cc @jonahkagan as our resident CDF export to seed the idea!

And the migration path from the current approach to option 1 above feels like a straightforward one, so should be easy if/when the time comes.

[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/types/src/ballot_package.ts:20]: I think what you've set up here is a good solution for now. I had some other ideas for handling the other custom fields that we currently have, but I agree that if we need to add more custom fields in the future we should consider having a separate file.

We will need to think about how this new file effects the election hash, which currently is just a hash of the election definition file.

**Pull request merged**